pkgs/applications/editors/vim/plugins/overrides.nix

```diff
@@ -341,6 +341,10 @@
     src = "${nodePackages."@yaegassy/coc-nginx"}/lib/node_modules/@yaegassy/coc-nginx";
   };
 
+  codecompanion-nvim = super.codecompanion-nvim.overrideAttrs {
+    dependencies = with self; [ curl plenary-nvim nvim-treesitter ];
```

Comment:
Not sure I follow -- why's that?

I've added a `nixpkgs-review` report and tested the plugin with my home-manager config against Ollama:
https://github.com/david-r-cox/nixos-config/compare/main...codecompanion.nvim

Both the plugin itself and text generation appear to be working as expected.

Reply:
@teto maybe we should replace all those patches with just adding the program to dependencies? Seems much better than patching (== providing debug hell for upstream devs) and much better UX than `passthru.initLua` (initLua is hard to find)

Reply:
After a discussion in Matrix, we decided to create a new attribute for such dependencies. Though for now there is not even a PR, so please use patching instead of `dependencies = [ curl ]`